src/estimator.cpp

```diff
@@ -181,30 +188,49 @@ void Estimator::run()
     // Below Eq. 45 Mahony Paper
     turbomath::Quaternion q_tilde = q_acc_inv * state_.attitude;
     // Correction Term of Eq. 47a and 47b Mahony Paper
-    // w_acc = 2*s_tilde*v_tilde
-    w_acc.x = -2.0f*q_tilde.w*q_tilde.x;
-    w_acc.y = -2.0f*q_tilde.w*q_tilde.y;
-    w_acc.z = 0.0f; // Don't correct z, because it's unobservable from the accelerometer
-
-    // integrate biases from accelerometer feedback
-    // (eq 47b Mahony Paper, using correction term w_acc found above
-    bias_.x -= ki*w_acc.x*dt;
-    bias_.y -= ki*w_acc.y*dt;
-    bias_.z = 0.0;  // Don't integrate z bias, because it's unobservable
-  }
-  else
-  {
-    w_acc.x = 0.0f;
-    w_acc.y = 0.0f;
-    w_acc.z = 0.0f;
+    // w_err = 2*s_tilde*v_tilde
+    w_err.x = -2.0f*q_tilde.w*q_tilde.x;
+    w_err.y = -2.0f*q_tilde.w*q_tilde.y;
+    w_err.z = 0.0f; // Don't correct z, because it's unobservable from the accelerometer
   }
 
   if (attitude_correction_next_run_)
   {
+    turbomath::Vector xhat_BW, yhat_BW, zhat_BW;
+    turbomath::Vector xext_BW, yext_BW, zext_BW;
+    {
+      float &w = state_.attitude.w, &x = state_.attitude.x, &y = state_.attitude.y, &z = state_.attitude.z;
+      xhat_BW.x = 1.0f - 2.0f*(y*y + z*z); xhat_BW.y = 2.0f*(x*y - z*w); xhat_BW.z = 2.0f*(x*z + y*w);
+      yhat_BW.x = 2.0f*(x*y + z*w); yhat_BW.y = 1.0f - 2.0f*(x*x + z*z); yhat_BW.z = 2.0f*(y*z - x*w);
+      zhat_BW.x = 2.0f*(x*z - y*w); zhat_BW.y = 2.0f*(y*z + x*w); zhat_BW.z = 1.0f - 2.0f*(x*x + y*y);
+    }
+    {
+      float &w = q_correction_.w, &x = q_correction_.x, &y = q_correction_.y, &z = q_correction_.z;
```

Comment:
Does it make sense to make this a function?

I actually think this could be a good opportunity to break up `estimator::run` so it's a little more concise and delegate the nitty-gritties to other functions.